### PR TITLE
Add ChainConfig to ethclient

### PR DESCRIPTION
**Pull request by cam-schultz**
## Why this should be merged
Adds `ChainConfig` to `ethclient.Client` so that it can be called by an external Go application. No namespace changes are required.

## How this works
See above.

## How this was tested
Local testing using an external Go application that calls the method via the client.

## How is this documented
N/A

### CODE REVIEW

**Review by ceyonur — APPROVED** [review]
lgtm

File: ethclient/ethclient.go [hunk 1]
```diff
@@ -141,6 +143,16 @@ func (ec *client) Client() *rpc.Client {
 
 // Blockchain Access
 
+// ChainConfig retrieves the current chain config.
+func (ec *client) ChainConfig(ctx context.Context) (*params.ChainConfigWithUpgradesJSON, error) {
+	var result *params.ChainConfigWithUpgradesJSON
```
Comment (aaronbuchwald) [inline ethclient/ethclient.go:148]: Can we return `*params.ChainConfig` instead of `*params.ChainConfigWithUpgradesJSON` since it's meant to be an intermediate type for marshalling?

Comment (cam-schultz) [reply ethclient/ethclient.go:148]: Sure thing. Should we also return `*params.ChainConfig` interally? https://github.com/ava-labs/subnet-evm/blob/master/internal/ethapi/api.go#L145

Comment (ceyonur) [reply ethclient/ethclient.go]: It's also used in `getChainConfig` API and it also returns `ChainConfigWithUpgradesJSON`. we should keep them same.

Comment (cam-schultz) [reply ethclient/ethclient.go]: Fixed now.

File: ethclient/ethclient.go [hunk 2]
```diff
@@ -141,6 +143,16 @@ func (ec *client) Client() *rpc.Client {
 
 // Blockchain Access
 
+// ChainConfig retrieves the current chain config.
+func (ec *client) ChainConfig(ctx context.Context) (*params.ChainConfig, error) {
```
Comment (ceyonur) [inline ethclient/ethclient.go:147]: I meant returning `ChainConfigWithUpgradesJSON` instead of  `ChainConfig` [here](https://github.com/ava-labs/subnet-evm/pull/1040#discussion_r1442719438). Sorry for the confusion.

File: internal/ethapi/api.go [hunk 3]
```diff
@@ -141,8 +141,8 @@ func (s *EthereumAPI) Syncing() (interface{}, error) {
 	return false, nil
 }
 
-func (s *BlockChainAPI) GetChainConfig(ctx context.Context) *params.ChainConfigWithUpgradesJSON {
-	return s.b.ChainConfig().ToWithUpgradesJSON()
+func (s *BlockChainAPI) GetChainConfig(ctx context.Context) *params.ChainConfig {
```
Comment (ceyonur) [inline internal/ethapi/api.go:144]: If we think it's more convenient to return this without Upgrades, we should add a new API method for `GetUpgrades`. Otherwise we should revert this change and use  `ChainConfigWithUpgradesJSON` in both client and API function. 
cc @aaronbuchwald

**Review by ceyonur — APPROVED** [review]

**Review by darioush — APPROVED** [review]
